clang/include/clang/Driver/Options.td

```diff
@@ -2074,6 +2074,32 @@ def fbounds_safety_debug_trap_reasons_EQ
     NormalizedValues<["None", "Basic", "Detailed"]>,
     MarshallingInfoEnum<CodeGenOpts<"BoundsSafetyDebugTrapReasons">, "Detailed">;
 
+def fbounds_safety_soft_traps_EQ
+  : Joined<["-"], "fbounds-safety-soft-traps=">, Group<f_Group>,
+    Visibility<[ClangOption, CC1Option]>,
+    HelpText<"Enables soft traps which causes the compiler to emit a non-fatal "
+      "instruction sequence when a bounds check fails. After this instruction "
+      "sequence execution resumes as if the bounds check had succeeded."
+      "For `call_with_*` modes see 'bounds_safety_soft_traps.h' for the runtime"
+      " function interface. This option supports the following modes: "
+      "`disabled` - No soft traps, i.e. use hard traps (default)."
+      "`call_with_str` - Emit a call to a runtime function that receives the "
+      "trap reason as a string."
+      "`call_with_code` - Emit a call to a runtime function that receives an "
+      "integer describing the trap reason. This is better for codesize than "
+      "`call_with_str` but is harder to debug if debug info is missing.">,
+    Values<"none,call_with_str,call_with_code">,
```

Comment:
Nit: other uses of `Values<>` in this file seem to use kebab-case rather than snake_case